build error: check_openblas_Apr2024.c is defined as an output multiple times

**Describe the bug**

I am getting this error while trying to compile the newest suitesparse 7.12:

```
2025-11-04T07:26:49.6457273Z -- ------------------------------------------------------------------------
2025-11-04T07:26:49.6457537Z -- SuiteSparse CMAKE report for: SuiteSparse
2025-11-04T07:26:49.6457775Z -- ------------------------------------------------------------------------
2025-11-04T07:26:49.6457990Z -- build type:           Release
2025-11-04T07:26:49.6458322Z -- BUILD_SHARED_LIBS:    ON
2025-11-04T07:26:49.6458492Z -- BUILD_STATIC_LIBS:    ON
2025-11-04T07:26:49.6458662Z -- C compiler:           /run/ccache/bin/cc 
2025-11-04T07:26:49.6459548Z -- C flags:              -O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -fdiagnostics-color=always
2025-11-04T07:26:49.6460450Z -- C++ compiler:         /run/ccache/bin/c++
2025-11-04T07:26:49.6461176Z -- C++ flags:            -O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -fdiagnostics-color=always
2025-11-04T07:26:49.6461904Z -- C Flags release:      -O3 -DNDEBUG 
2025-11-04T07:26:49.6462095Z -- C++ Flags release:    -O3 -DNDEBUG 
2025-11-04T07:26:49.6462296Z -- Fortran compiler:     /usr/bin/gfortran 
2025-11-04T07:26:49.6462490Z -- compile definitions:  
2025-11-04T07:26:49.6462699Z -- ------------------------------------------------------------------------
2025-11-04T07:26:49.6472195Z -- Configuring done (4.8s)
2025-11-04T07:26:49.8693611Z [31mCMake Error:
2025-11-04T07:26:49.8693776Z   Running
2025-11-04T07:26:49.8693887Z 
2025-11-04T07:26:49.8694057Z    '/usr/bin/ninja' '-C' '/run/build/SuiteSparse/_flatpak_build' '-t' 'recompact'
2025-11-04T07:26:49.8694265Z 
2025-11-04T07:26:49.8694328Z   failed with:
2025-11-04T07:26:49.8694409Z 
2025-11-04T07:26:49.8694759Z    ninja: error: build.ninja:39116: /run/build/SuiteSparse/SuiteSparse_config/cmake_modules/check_openblas_Apr2024.c is defined as an output multiple times
2025-11-04T07:26:49.8695150Z 
2025-11-04T07:26:49.8695152Z 
2025-11-04T07:26:49.8695155Z 
2025-11-04T07:26:49.8695157Z 
2025-11-04T07:26:49.8695207Z [0m
```

**To Reproduce**

Trigger build on https://github.com/flathub/org.gimp.GIMP/pull/554

**Expected behavior**

Suitesparse builds as it was on 7.11

**Desktop (please complete the following information):**
 - OS: None, Flatpak
 - compiler: 15.2.0
 - Version: 7.12.0

--- a/SuiteSparse_config/cmake_modules/SuiteSparse__blas_threading.cmake
+++ b/SuiteSparse_config/cmake_modules/SuiteSparse__blas_threading.cmake
@@ -40,9 +40,12 @@ if ( NOT ( ${CMAKE_SYSTEM} MATCHES ${CMAKE_HOST_SYSTEM} ) )
 
     if ( ${BLA_VENDOR} STREQUAL "OpenBLAS" )
         # check if OpenBLAS has openblas_set_num_threads_local (Apr 2024 or later)
+        get_filename_component ( ABS_SOURCE_PATH
+            ${PROJECT_SOURCE_DIR}/../SuiteSparse_config/cmake_modules/check_openblas_Apr2024.c
+            ABSOLUTE )
         try_compile ( OPENBLAS_2024_COMPILES
             ${CMAKE_CURRENT_BINARY_DIR}
-            ${PROJECT_SOURCE_DIR}/../SuiteSparse_config/cmake_modules/check_openblas_Apr2024.c
+            ${ABS_SOURCE_PATH}
             LINK_OPTIONS    ${BLAS_LINKER_FLAGS}
             LINK_LIBRARIES  ${BLAS_LIBRARIES} )
         if ( ${OPENBLAS_2024_COMPILES} )
@@ -69,9 +72,12 @@ if ( is_Intel )
     # determine if MKL is single-threaded or multi-threaded (info only)
     #---------------------------------------------------------------------------
 
+    get_filename_component ( ABS_SOURCE_PATH
+        ${PROJECT_SOURCE_DIR}/../SuiteSparse_config/cmake_modules/check_mkl.c
+        ABSOLUTE )
     try_run ( MKL_RUNS MKL_COMPILES
         ${CMAKE_CURRENT_BINARY_DIR}
-        ${PROJECT_SOURCE_DIR}/../SuiteSparse_config/cmake_modules/check_mkl.c
+        ${PROJECT_SOURCE_DIR}
         LINK_OPTIONS    ${BLAS_LINKER_FLAGS}
         LINK_LIBRARIES  ${BLAS_LIBRARIES}
         RUN_OUTPUT_VARIABLE MKL_OUTPUT )
@@ -99,9 +105,12 @@ endif ( )
 if ( ${BLA_VENDOR} STREQUAL "OpenBLAS" )
 
     # check if OpenBLAS has openblas_get_num_threads (Mar 2015 or later)
+    get_filename_component ( ABS_SOURCE_PATH
+        ${PROJECT_SOURCE_DIR}/../SuiteSparse_config/cmake_modules/check_openblas_Mar2015.c
+        ABSOLUTE )
     try_run ( OPENBLAS_2015_RUNS OPENBLAS_2015_COMPILES
         ${CMAKE_CURRENT_BINARY_DIR}
-        ${PROJECT_SOURCE_DIR}/../SuiteSparse_config/cmake_modules/check_openblas_Mar2015.c
+        ${ABS_SOURCE_PATH}
         LINK_OPTIONS    ${BLAS_LINKER_FLAGS}
         LINK_LIBRARIES  ${BLAS_LIBRARIES}
         RUN_OUTPUT_VARIABLE OPENBLAS_2015_OUTPUT )
@@ -115,9 +124,12 @@ if ( ${BLA_VENDOR} STREQUAL "OpenBLAS" )
     endif ( )
 
     # check if OpenBLAS has openblas_set_num_threads_local (Apr 2024 or later)
+    get_filename_component ( ABS_SOURCE_PATH
+        ${PROJECT_SOURCE_DIR}/../SuiteSparse_config/cmake_modules/check_openblas_Apr2024.c
+        ABSOLUTE )
     try_run ( OPENBLAS_2024_RUNS OPENBLAS_2024_COMPILES
         ${CMAKE_CURRENT_BINARY_DIR}
-        ${PROJECT_SOURCE_DIR}/../SuiteSparse_config/cmake_modules/check_openblas_Apr2024.c
+        ${ABS_SOURCE_PATH}
         LINK_OPTIONS    ${BLAS_LINKER_FLAGS}
         LINK_LIBRARIES  ${BLAS_LIBRARIES}
         RUN_OUTPUT_VARIABLE OPENBLAS_2024_OUTPUT )

